src/xpk/core/storage.py

```diff
@@ -0,0 +1,340 @@
+    self.readonly: bool = spec.get("readonly")
+    self.manifest: str = spec.get("manifest")
+    self.pvc: str = spec.get("pvc")
+    self.pv: str = spec.get("pv")
+    self.bucket: str = self._get_bucket()
+
+  def fields_as_list(self) -> list[str]:
+    """
+    Returns a list of fields for display purposes.
+
+    Returns:
+        A list of strings representing the Storage object's fields.
+    """
+    return [
+        self.name,
+        self.type,
+        self.auto_mount,
+        self.mount_point,
+        self.readonly,
+        self.manifest,
+    ]
+
+  def _get_bucket(self) -> str:
+    """
+    Retrieves the bucket name from PersistentVolume definition associated with the storage.
+
+    Returns:
+        The name of the bucket.
+    """
+    client = k8s_client.CoreV1Api()
+    try:
+      pv: V1PersistentVolume = client.read_persistent_volume(self.pv)
+    except client.ApiException as e:
+      xpk_print(
+          f"Exception when calling CoreV1Api->read_persistent_volume: {e}"
+      )
+    return pv.spec.csi.volume_handle
+
+  def get_mount_options(self) -> list[str]:
+    """
+    Retrieves the mount options for the PersistentVolume.
+
+    Returns:
+        A list of mount options.
+    """
+    client = k8s_client.CoreV1Api()
+    try:
+      pv: V1PersistentVolume = client.read_persistent_volume(self.pv)
+    except client.ApiException as e:
+      xpk_print(
+          f"Exception when calling CoreV1Api->read_persistent_volume: {e}"
+      )
+    return pv.spec.mount_options
+
+
+def list_storages(k8s_api_client: ApiClient) -> list[Storage]:
+  """
+  Lists all Storage custom resources in the cluster.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+
+  Returns:
+      A list of Storage objects representing the Storage resources.
+  """
+  api_instance = k8s_client.CustomObjectsApi(k8s_api_client)
+  try:
+    resp = api_instance.list_cluster_custom_object(
+        group=XPK_API_GROUP_NAME,
+        version=XPK_API_GROUP_VERSION,
+        plural=STORAGE_CRD_KIND.lower() + "s",
+    )
+  except ApiException as e:
+    xpk_print(f"Kubernetes API exception while listing Storages: {e}")
+    xpk_exit(1)
+
+  storages = []
+  for stg in resp["items"]:
+    storage = Storage(stg)
+    storages.append(storage)
+  return storages
+
+
+def get_auto_mount_storages(k8s_api_client: ApiClient) -> list[Storage]:
+  """
+  Retrieves all Storage resources that have --auto-mount flag set to true.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+
+  Returns:
+      A list of Storage objects that have `auto_mount` set to True.
+  """
+  auto_mount_storages: list[Storage] = []
+  for storage in list_storages(k8s_api_client):
+    if storage.auto_mount is True:
+      auto_mount_storages.append(storage)
+  return auto_mount_storages
+
+
+def get_storages(k8s_api_client: ApiClient, names: list[str]) -> list[Storage]:
```

Comment:
The naming for this method can be more descriptive. We have a method `get_storage` that retrieves a single storage. Therefore, we expect if get_storages retrieves the list of storage objects based on names list passed in, but it also add some other "auto_mount" objects here.

I would suggest to
- keep this `get_storages` method for the names only.
- add a new method `get_storages_to_mount` that unions the data from `get_storages` and `get_auto_mount_storages`

Reply:
Good point, thanks. Done!